core/src/shapes/pax_rects.c

```diff
@@ -407,6 +407,16 @@ void pax_shade_rect(
         width         = tmp.w;
         height        = tmp.h;
 
+        // Normalize negative dimensions after orientation transform
```

Comment:
This should have already been handled by `pax_orient_det_rectf`; either remove this or apply any fixes in there.